dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SynchronizationController.java

```diff
@@ -68,29 +64,21 @@
     @Autowired
     private RestTemplate restTemplate;
 
-    @Autowired
-    private RenderService renderService;
-
     @PreAuthorize( "hasRole('ALL') or hasRole('F_EXPORT_DATA')" )
-    @PostMapping( "/dataPush" )
-    public void execute( HttpServletResponse response )
+    @PostMapping( value = "/dataPush", produces = APPLICATION_JSON_VALUE )
+    @ResponseBody
+    public ImportConflicts execute()
         throws IOException
     {
-        ImportConflicts summary = synchronizationManager.executeDataValuePush();
-
-        response.setContentType( CONTENT_TYPE_JSON );
-        renderService.toJson( response.getOutputStream(), summary );
+        return synchronizationManager.executeDataValuePush();
     }
 
     @PreAuthorize( "hasRole('ALL')" )
-    @PostMapping( "/metadataPull" )
-    public void importMetaData( @RequestBody String url, HttpServletResponse response )
-        throws IOException
+    @PostMapping( value = "/metadataPull", produces = APPLICATION_JSON_VALUE )
+    @ResponseBody
+    public ImportReport importMetaData( @RequestBody String url )
```

Comment:
FYI: This method could not be tested meaningfully with spring MVC tests as it would always throw an exception unless we would provide it with an `url` actual an actual metadata endpoint. Please look extra careful here.